### PR TITLE
[bitnami/postgresql-ha] Add psql query to postgresql container liveness probe

### PR DESCRIPTION
**Pull request by dani8art**
**Description of the change**

This PR adds a better approach to the PostgreSQL container liveness probe. This new approach is just a `psql` query that replaces the `pg_isready`.

**Benefits**

Executing a `psql` query would be a better liveness probe since it is doing a real query against the database. 

**Possible drawbacks**

This specific query `SELECT 1;` could not be enough and it could be a flaky test. 

**Applicable issues**

N/A

**Additional information**

I have updated the `_helpers.tpl` in order to solve a bug. When we use `if not (empty (include "postgresql-ha.postgresqlDatabase" .))` I realize that if we use an `include` inside an `empty` we should take care about returning a `null` value and we always must return `""`. If not the empty will always return `true`. 

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/postgresql-ha/templates/_helpers.tpl [hunk 1]
```diff
@@ -314,18 +314,19 @@ Helm 2.11 supports the assignment of a value to a variable defined in a differen
 but Helm 2.9 and 2.10 doesn't support it, so we need to implement this if-else logic.
 Also, we can't use a single if because lazy evaluation is not an option
 */}}
+{{- $postgresqlDatabase := default "postgres" .Values.postgresql.database -}}
 {{- if .Values.global }}
     {{- if .Values.global.postgresql }}
         {{- if .Values.global.postgresql.database }}
-            {{- .Values.global.postgresql.database -}}
+            {{- default "" .Values.global.postgresql.database -}}
```
Comment (alemorcuq) [inline bitnami/postgresql-ha/templates/_helpers.tpl]:
```suggestion
            {{- default "postgres" .Values.global.postgresql.database -}}
```

File: bitnami/postgresql-ha/templates/_helpers.tpl [hunk 2]
```diff
@@ -734,3 +735,12 @@ postgresql-ha: Upgrade repmgr extension
       --set postgresql.upgradeRepmgrExtension=true
 {{- end -}}
 {{- end -}}
+
+{{/* Set PGPASSWORD as environment variable depends on configuration */}}
+{{- define "postgresql-ha.pgpassword" -}}
+{{- if .Values.postgresql.usePasswordFile -}}
+PGPASSWORD=$(cat $POSTGRES_PASSWORD_FILE)
```
Comment (alemorcuq) [inline bitnami/postgresql-ha/templates/_helpers.tpl]:
```suggestion
PGPASSWORD=$(< $POSTGRES_PASSWORD_FILE)
```

File: bitnami/postgresql-ha/templates/postgresql/statefulset.yaml [hunk 3]
```diff
@@ -150,11 +150,7 @@ spec:
               command:
                 - sh
                 - -c
-                {{- if not (empty (include "postgresql-ha.postgresqlDatabase" .)) }}
-                - exec pg_isready -U {{ include "postgresql-ha.postgresqlUsername" . | quote }} -d {{ (include "postgresql-ha.postgresqlDatabase" .) | quote }} -h 127.0.0.1
-                {{- else }}
-                - exec pg_isready -U {{ include "postgresql-ha.postgresqlUsername" . | quote }} -h 127.0.0.1
-                {{- end }}
+                - '{{include "postgresql-ha.pgpassword" . }} psql -w -U {{ include "postgresql-ha.postgresqlUsername" . | quote }} -d {{ (include "postgresql-ha.postgresqlDatabase" .) | quote }}  -h 127.0.0.1 -c "SELECT 1"'
```
Comment (alemorcuq) [inline bitnami/postgresql-ha/templates/postgresql/statefulset.yaml]:
```suggestion
                - '{{ include "postgresql-ha.pgpassword" . }} psql -w -U {{ include "postgresql-ha.postgresqlUsername" . | quote }} -d {{ include "postgresql-ha.postgresqlDatabase" . | quote }} -h 127.0.0.1 -c "SELECT 1"'
```

File: bitnami/postgresql-ha/templates/postgresql/statefulset.yaml [hunk 4]
```diff
@@ -167,11 +163,7 @@ spec:
...
```
Comment (alemorcuq) [inline bitnami/postgresql-ha/templates/postgresql/statefulset.yaml]:
```suggestion
                - '{{ include "postgresql-ha.pgpassword" . }} psql -w -U {{ include "postgresql-ha.postgresqlUsername" . | quote }} -d {{ include "postgresql-ha.postgresqlDatabase" . | quote }} -h 127.0.0.1 -c "SELECT 1"'
```

File: bitnami/postgresql-ha/templates/_helpers.tpl [hunk 5]
```diff
@@ -734,3 +735,12 @@ postgresql-ha: Upgrade repmgr extension
...
+{{- else -}}
+PGPASSWORD=$POSTGRES_PASSWORD
+{{- end -}}
```
Comment (alemorcuq) [inline bitnami/postgresql-ha/templates/_helpers.tpl]: I'm not sure about this depending on environment variables. Can we make it only dependant on the values itself?

File: bitnami/postgresql-ha/templates/_helpers.tpl (hunk 1 above)
Comment (dani8art) [reply bitnami/postgresql-ha/templates/_helpers.tpl]: done

File: bitnami/postgresql-ha/templates/_helpers.tpl (hunk 2 above)
Comment (dani8art) [reply bitnami/postgresql-ha/templates/_helpers.tpl]: done

File: bitnami/postgresql-ha/templates/postgresql/statefulset.yaml (hunk 3 above)
Comment (dani8art) [reply bitnami/postgresql-ha/templates/postgresql/statefulset.yaml]: done

File: bitnami/postgresql-ha/templates/postgresql/statefulset.yaml (hunk 4 above)
Comment (dani8art) [reply bitnami/postgresql-ha/templates/postgresql/statefulset.yaml]: done

**Pull request merged**